File: scenarios/basic/desktop/linux-qt/SmallTouchTarget.py
import sys
from PyQt5.QtWidgets import (QApplication, QMainWindow, QVBoxLayout, QHBoxLayout, 
                             QWidget, QLabel, QPushButton, QFrame, QScrollArea)
from PyQt5.QtCore import Qt
from PyQt5.QtGui import QFont

class SmallTouchTarget(QMainWindow):
    def __init__(self):
        super().__init__()
        self.setWindowTitle("Social Media Feed")
        self.setGeometry(100, 100, 600, 800)
        
        # Central widget
        central_widget = QWidget()
        self.setCentralWidget(central_widget)
        
        # Main layout
        main_layout = QVBoxLayout(central_widget)
        main_layout.setSpacing(0)
        
        # Tab navigation
        tab_layout = QHBoxLayout()
        tab_layout.setSpacing(0)
        
        following_button = QPushButton("Following")
        following_button.setStyleSheet("QPushButton { background-color: #007bff; color: white; font-weight: bold; }")
        following_button.setFixedHeight(50)
        tab_layout.addWidget(following_button)
        
        discover_button = QPushButton("Discover")
        discover_button.setFixedHeight(50)
        tab_layout.addWidget(discover_button)
        
        trending_button = QPushButton("Trending")
        trending_button.setFixedHeight(50)
        tab_layout.addWidget(trending_button)
        
        main_layout.addLayout(tab_layout)
        
        # Posts scrollable area
        scroll_area = QScrollArea()
        scroll_area.setWidgetResizable(True)
        scroll_area.setVerticalScrollBarPolicy(Qt.ScrollBarAsNeeded)
        
        posts_widget = QWidget()
        posts_layout = QVBoxLayout(posts_widget)
        posts_layout.setSpacing(15)
        posts_layout.setContentsMargins(20, 20, 20, 20)
        
        # Load posts
        self.load_posts(posts_layout)
        
        scroll_area.setWidget(posts_widget)
        main_layout.addWidget(scroll_area)
        
    def load_posts(self, layout):
        posts = [
            {
                "username": "john_doe",
                "time_ago": "2h",
                "content": "Beautiful sunset from my hike today! 🏔️ #nature #photography",
                "likes": 42,
                "comments": 8,
                "is_liked": False,
                "is_bookmarked": False
            },
            {
                "username": "sarah_j",
                "time_ago": "4h",
                "content": "Just finished reading an amazing book! Highly recommend it to everyone.",
                "likes": 28,
                "comments": 12,
                "is_liked": False,
                "is_bookmarked": False
            },
            {
                "username": "mike_chen",
                "time_ago": "6h",
                "content": "Working on a new project. Excited to share the results soon!",
                "likes": 15,
                "comments": 3,
                "is_liked": False,
                "is_bookmarked": False
            }
        ]
        
        for post in posts:
            post_frame = self.create_post_card(post)
            layout.addWidget(post_frame)
    
    def create_post_card(self, post):
        # Post frame
        frame = QFrame()
        frame.setFrameStyle(QFrame.Box)
        frame.setStyleSheet("QFrame { background-color: white; border: 1px solid #ddd; }")
        
        main_layout = QVBoxLayout(frame)
        main_layout.setSpacing(10)
        main_layout.setContentsMargins(15, 15, 15, 15)
        
        # Profile section
        profile_layout = QHBoxLayout()
        
        # Profile image
        profile_image = QLabel("👤")
        profile_image.setFixedSize(40, 40)
        profile_image.setStyleSheet("font-size: 24px;")
        profile_layout.addWidget(profile_image)
        
        # Username and time
        user_info_layout = QVBoxLayout()
        
        username_label = QLabel(post['username'])
        username_font = QFont()
        username_font.setBold(True)
        username_label.setFont(username_font)
        user_info_layout.addWidget(username_label)
        
        time_label = QLabel(post['time_ago'])
        user_info_layout.addWidget(time_label)
        
        profile_layout.addLayout(user_info_layout)
        
        # SMALL TOUCH TARGET - Only 20x20px on mobile interface
        more_button = QPushButton("⋯")
        more_button.setFixedSize(20, 20)  # SMALL TOUCH TARGET
        more_button.setStyleSheet("QPushButton { background-color: #e0e0e0; border-radius: 4px; }")
        profile_layout.addWidget(more_button)
        
        main_layout.addLayout(profile_layout)
        
        # Post image
        post_image = QLabel("🖼️")
        post_image.setAlignment(Qt.AlignCenter)
        post_image.setFixedHeight(200)
        post_image.setStyleSheet("font-size: 100px; background-color: #f0f0f0;")
        main_layout.addWidget(post_image)
        
        # Post actions
        actions_layout = QHBoxLayout()
        actions_layout.setSpacing(16)
        
        # SMALL TOUCH TARGET - Only 20x20px
        like_button = QPushButton("♡" if not post['is_liked'] else "♥")
        like_button.setFixedSize(20, 20)  # SMALL TOUCH TARGET
        like_button.setStyleSheet("QPushButton { background-color: #e0e0e0; border-radius: 4px; }")
        actions_layout.addWidget(like_button)
        
        # SMALL TOUCH TARGET - Only 20x20px
        comment_button = QPushButton("💬")
        comment_button.setFixedSize(20, 20)  # SMALL TOUCH TARGET
        comment_button.setStyleSheet("QPushButton { background-color: #e0e0e0; border-radius: 4px; }")
        actions_layout.addWidget(comment_button)
        
        # SMALL TOUCH TARGET - Only 20x20px
        share_button = QPushButton("↗️")
        share_button.setFixedSize(20, 20)  # SMALL TOUCH TARGET
        share_button.setStyleSheet("QPushButton { background-color: #e0e0e0; border-radius: 4px; }")
        actions_layout.addWidget(share_button)
        
        # SMALL TOUCH TARGET - Only 20x20px
        bookmark_button = QPushButton("🔖")
        bookmark_button.setFixedSize(20, 20)  # SMALL TOUCH TARGET
        bookmark_button.setStyleSheet("QPushButton { background-color: #e0e0e0; border-radius: 4px; }")
        actions_layout.addWidget(bookmark_button)
        
        main_layout.addLayout(actions_layout)
        
        # Likes count
        likes_label = QLabel(f"{post['likes']} likes")
        likes_font = QFont()
        likes_font.setBold(True)
        likes_label.setFont(likes_font)
        main_layout.addWidget(likes_label)
        
        # Post content
        content_label = QLabel(f"{post['username']} {post['content']}")
        content_font = QFont()
        content_font.setBold(True)
        content_label.setFont(content_font)
        content_label.setWordWrap(True)
        main_layout.addWidget(content_label)
        
        # Comments link
        comments_label = QLabel(f"View all {post['comments']} comments")
        comments_label.setStyleSheet("color: #666;")
        main_layout.addWidget(comments_label)
        
        return frame

def main():
    app = QApplication(sys.argv)
    window = SmallTouchTarget()
    window.show()
    sys.exit(app.exec_())

if __name__ == "__main__":
    main()

File: scenarios/basic/cross-platform/qt/SmallTouchTarget.cpp
#include <QApplication>
#include <QMainWindow>
#include <QVBoxLayout>
#include <QHBoxLayout>
#include <QGridLayout>
#include <QLabel>
#include <QPushButton>
#include <QWidget>
#include <QScrollArea>
#include <QFrame>

class SocialMediaApp : public QMainWindow
{
    Q_OBJECT

public:
    SocialMediaApp(QWidget *parent = nullptr) : QMainWindow(parent)
    {
        setupUI();
        setupConnections();
    }

private slots:
    void likePost()
    {
        QMessageBox::information(this, "Like", "Post liked!");
    }
    
    void sharePost()
    {
        QMessageBox::information(this, "Share", "Post shared!");
    }
    
    void commentPost()
    {
        QMessageBox::information(this, "Comment", "Comment added!");
    }
    
    void followUser()
    {
        QMessageBox::information(this, "Follow", "User followed!");
    }

private:
    void setupUI()
    {
        QWidget *centralWidget = new QWidget(this);
        setCentralWidget(centralWidget);
        
        QVBoxLayout *mainLayout = new QVBoxLayout(centralWidget);
        
        // Header
        QHBoxLayout *headerLayout = new QHBoxLayout();
        
        QLabel *titleLabel = new QLabel("SocialFeed", this);
        titleLabel->setStyleSheet(
            "QLabel {"
            "    font-size: 24px;"
            "    font-weight: bold;"
            "    color: #333;"
            "    padding: 15px 0;"
            "}"
        );
        headerLayout->addWidget(titleLabel);
        
        headerLayout->addStretch();
        
        QPushButton *profileBtn = new QPushButton("👤 Profile", this);
        profileBtn->setStyleSheet(
            "QPushButton {"
            "    background: #007bff;"
            "    color: white;"
            "    border: none;"
            "    border-radius: 6px;"
            "    padding: 8px 16px;"
            "    font-size: 14px;"
            "    font-weight: bold;"
            "}"
            "QPushButton:hover {"
            "    background: #0056b3;"
            "}"
        );
        headerLayout->addWidget(profileBtn);
        
        mainLayout->addLayout(headerLayout);
        
        // Feed content
        QScrollArea *scrollArea = new QScrollArea(this);
        scrollArea->setWidgetResizable(true);
        scrollArea->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
        
        QWidget *scrollWidget = new QWidget();
        QVBoxLayout *feedLayout = new QVBoxLayout(scrollWidget);
        feedLayout->setSpacing(20);
        
        // Post 1
        QFrame *post1 = createPost(
            "John Doe",
            "2 hours ago",
            "Just finished an amazing hike in the mountains! The views were incredible. #hiking #nature #adventure",
            "🏔️"
        );
        feedLayout->addWidget(post1);
        
        // Post 2
        QFrame *post2 = createPost(
            "Sarah Wilson",
            "4 hours ago",
            "Working on a new project. Can't wait to share the results with everyone! #work #project #excited",
            "💻"
        );
        feedLayout->addWidget(post2);
        
        // Post 3
        QFrame *post3 = createPost(
            "Mike Johnson",
            "6 hours ago",
            "Delicious dinner at the new restaurant downtown. Highly recommend! #food #restaurant #dinner",
            "🍽️"
        );
        feedLayout->addWidget(post3);
        
        // Post 4
        QFrame *post4 = createPost(
            "Emma Davis",
            "8 hours ago",
            "Beautiful sunset from my balcony tonight. Sometimes the simple things are the most beautiful. #sunset #beauty #peaceful",
            "🌅"
        );
        feedLayout->addWidget(post4);
        
        scrollArea->setWidget(scrollWidget);
        mainLayout->addWidget(scrollArea);
    }
    
    QFrame* createPost(const QString &author, const QString &time, const QString &content, const QString &icon)
    {
        QFrame *post = new QFrame(this);
        post->setStyleSheet(
            "QFrame {"
            "    background: white;"
            "    border: 1px solid #ddd;"
            "    border-radius: 8px;"
            "    padding: 20px;"
            "    margin: 10px 0;"
            "}"
        );
        
        QVBoxLayout *postLayout = new QVBoxLayout(post);
        postLayout->setSpacing(15);
        
        // Author info
        QHBoxLayout *authorLayout = new QHBoxLayout();
        
        QLabel *authorIcon = new QLabel(icon, this);
        authorIcon->setStyleSheet(
            "QLabel {"
            "    font-size: 24px;"
            "    margin-right: 10px;"
            "}"
        );
        authorLayout->addWidget(authorIcon);
        
        QVBoxLayout *authorInfo = new QVBoxLayout();
        
        QLabel *authorName = new QLabel(author, this);
        authorName->setStyleSheet(
            "QLabel {"
            "    font-size: 16px;"
            "    font-weight: bold;"
            "    color: #333;"
            "}"
        );
        authorInfo->addWidget(authorName);
        
        QLabel *postTime = new QLabel(time, this);
        postTime->setStyleSheet(
            "QLabel {"
            "    font-size: 12px;"
            "    color: #666;"
            "}"
        );
        authorInfo->addWidget(postTime);
        
        authorLayout->addLayout(authorInfo);
        authorLayout->addStretch();
        
        QPushButton *followBtn = new QPushButton("Follow", this);
        followBtn->setStyleSheet(
            "QPushButton {"
            "    background: #28a745;"
            "    color: white;"
            "    border: none;"
            "    border-radius: 4px;"
            "    padding: 6px 12px;"
            "    font-size: 12px;"
            "    font-weight: bold;"
            "}"
            "QPushButton:hover {"
            "    background: #218838;"
            "}"
        );
        connect(followBtn, &QPushButton::clicked, this, &SocialMediaApp::followUser);
        authorLayout->addWidget(followBtn);
        
        postLayout->addLayout(authorLayout);
        
        // Post content
        QLabel *postContent = new QLabel(content, this);
        postContent->setStyleSheet(
            "QLabel {"
            "    font-size: 14px;"
            "    color: #333;"
            "    line-height: 1.5;"
            "}"
        );
        postContent->setWordWrap(true);
        postLayout->addWidget(postContent);
        
        // Action buttons
        QHBoxLayout *actionLayout = new QHBoxLayout();
        
        // SMALL TOUCH TARGET - Social share button only 20x20px on mobile interface
        QPushButton *likeBtn = new QPushButton("❤️", this);
        likeBtn->setStyleSheet(
            "QPushButton {"
            "    background: #f8f9fa;"
            "    border: 1px solid #ddd;"
            "    border-radius: 4px;"
            "    font-size: 14px;"
            "    min-width: 20px;"  // SMALL TOUCH TARGET - Only 20px wide
            "    min-height: 20px;" // SMALL TOUCH TARGET - Only 20px high
            "    max-width: 20px;"
            "    max-height: 20px;"
            "}"
            "QPushButton:hover {"
            "    background: #e9ecef;"
            "}"
        );
        connect(likeBtn, &QPushButton::clicked, this, &SocialMediaApp::likePost);
        actionLayout->addWidget(likeBtn);
        
        // SMALL TOUCH TARGET - Social share button only 20x20px on mobile interface
        QPushButton *commentBtn = new QPushButton("💬", this);
        commentBtn->setStyleSheet(
            "QPushButton {"
            "    background: #f8f9fa;"
            "    border: 1px solid #ddd;"
            "    border-radius: 4px;"
            "    font-size: 14px;"
            "    min-width: 20px;"  // SMALL TOUCH TARGET - Only 20px wide
            "    min-height: 20px;" // SMALL TOUCH TARGET - Only 20px high
            "    max-width: 20px;"
            "    max-height: 20px;"
            "}"
            "QPushButton:hover {"
            "    background: #e9ecef;"
            "}"
        );
        connect(commentBtn, &QPushButton::clicked, this, &SocialMediaApp::commentPost);
        actionLayout->addWidget(commentBtn);
        
        // SMALL TOUCH TARGET - Social share button only 20x20px on mobile interface
        QPushButton *shareBtn = new QPushButton("📤", this);
        shareBtn->setStyleSheet(
            "QPushButton {"
            "    background: #f8f9fa;"
            "    border: 1px solid #ddd;"
            "    border-radius: 4px;"
            "    font-size: 14px;"
            "    min-width: 20px;"  // SMALL TOUCH TARGET - Only 20px wide
            "    min-height: 20px;" // SMALL TOUCH TARGET - Only 20px high
            "    max-width: 20px;"
            "    max-height: 20px;"
            "}"
            "QPushButton:hover {"
            "    background: #e9ecef;"
            "}"
        );
        connect(shareBtn, &QPushButton::clicked, this, &SocialMediaApp::sharePost);
        actionLayout->addWidget(shareBtn);
        
        actionLayout->addStretch();
        
        QLabel *likesLabel = new QLabel("12 likes", this);
        likesLabel->setStyleSheet(
            "QLabel {"
            "    font-size: 12px;"
            "    color: #666;"
            "}"
        );
        actionLayout->addWidget(likesLabel);
        
        postLayout->addLayout(actionLayout);
        
        return post;
    }
    
    void setupConnections()
    {
        // Additional connections can be added here
    }
};

int main(int argc, char *argv[])
{
    QApplication app(argc, argv);
    
    SocialMediaApp app;
    app.setWindowTitle("SocialFeed - Social Media App");
    app.resize(600, 800);
    app.show();
    
    return app.exec();
}

#include "SmallTouchTarget.moc"
